### ISSUE TITLE
Bundle calls for principal property requests

### ISSUE THREAD

**Issue by Defcon0** [nextcloud/calendar#2030]
**Is your feature request related to a problem? Please describe.**

We have the tool installed for our company with about 65 calendars. When simply opening the calendar month view, the application calls 65 XHR requests simultaneously (URL is https://cloud.heimrich-hannot.com/remote.php/dav/principals/users/john@doe.de/). Even the inactive calendars which should be affecting the month view anyway are called. This leads to a website taking about 15 seconds to be completely loaded.

**Describe the solution you'd like**
Don't do the ajax calls for **inactive** calendars.

**Additional context**
Nextcloud: 17.0.3
Calendar: 2.0.1

**Comment by georgehrke** [comment]
![7A6A8662-8BE1-4E63-A4CE-FF24A66FBEEB](https://user-images.githubusercontent.com/1250540/75435543-764ae980-5953-11ea-9ae9-576d75147cc9.png)

Can't reproduce that behaviour. I have 20 calendars in the development instance. 
The first three requests you see are for building up the CalDAV connection.
The last two are for querying the two active calendars.
There are no requests for the 18 disiabled calendars.

**Comment by Defcon0** [comment]
Are you testing version 2.0.1 or the master branch? I think before the last update we hadn't had the problem (we had 1.7.1 before).

**Comment by Defcon0** [comment]
And could you please provide a screenshot how your User Interface looks like? Maybe we somehow added the calendar the wrong way?

**Comment by georgehrke** [comment]
I tested my local development instance with master as well as a production instance with 2.0.1. 
Both work as expected.

**Comment by tcitworld** [comment]
And I confirm it works on NC 17.0.3 with Calendar app 2.0.1 as well.

**Comment by tcitworld** [comment]
However, it indeed does requests to `https://framagenda.org/remote.php/dav/principals/users/%user%/` even if the calendar is disabled.

**Comment by georgehrke** [comment]
Yes, those are principal property requests to gather information like Displayname, etc.
That's necessary for displaying the avatar.

**Comment by Defcon0** [comment]
I understand, but this results in major performance issues for larger companies :-( Is there any way to deactivate that handling?

**Comment by georgehrke** [comment]
I'm personally using the calendar app in multiple major instances with more than one hundred calendars (and I'm aware of many others who do so too), where this is not causing any issues, so this seems to be some setup issue to me.

---

If you wish support with setup issues from Nextcloud GmbH we offer this as part of the Nextcloud subscription. Learn more about this at https://nextcloud.com/enterprise/

**Comment by Defcon0** [comment]
In my opinion this is a flaw in design. Of course, if I throw faster hardware on a problem, it's "fixed", but indeed the application could be much more efficient in this point if those information would be gathered with one single request and not hundreds running simultaneously.

**Comment by georgehrke** [comment]
If you think this is a design flaw, not a setup issue, a pull-request would be very much appreciated :)

**Comment by Defcon0** [comment]
@tcitworld What's your opinion on that?

Concerning a PR: I don't think, I'm able to do so I'm afraid due to time and knowledge constraints :-(

**Comment by SebastianKrupinski** [comment]
Closing this as this seems to be fixed, from the comments above and tests
